cfme/tests/intelligence/reports/test_validate_chargeback_report.py

```diff
@@ -466,32 +467,47 @@ def chargeback_report_custom(appliance, vm_ownership, assign_custom_rate, provid
     report.queue(wait_for_finish=True)
 
     yield list(report.saved_reports.all()[0].data.rows)
-    report.delete()
+
+    if report.exists:
+        report.delete()
 
 
 @pytest.fixture(scope="module")
-def new_compute_rate():
+def new_compute_rate(appliance):
     # Create a new Compute Chargeback rate
     try:
-        desc = 'custom_' + fauxfactory.gen_alphanumeric()
-        compute = rates.ComputeRate(description=desc,
-                    fields={'Used CPU':
-                            {'per_time': 'Hourly', 'variable_rate': '3'},
-                            'Used Disk I/O':
-                            {'per_time': 'Hourly', 'variable_rate': '2'},
-                            'Used Memory':
-                            {'per_time': 'Hourly', 'variable_rate': '2'}})
-        compute.create()
-        if not BZ(1532368, forced_streams=['5.9']).blocks:
-            storage = rates.StorageRate(description=desc,
-                    fields={'Used Disk Storage':
-                            {'per_time': 'Hourly', 'variable_rate': '3'}})
-            storage.create()
-        yield desc
-    finally:
-        compute.delete()
-        if not BZ(1532368, forced_streams=['5.9']).blocks:
-            storage.delete()
+        desc = 'cstm_' + fauxfactory.gen_alphanumeric()
+        compute = appliance.collections.compute_rates.create(
+            description=desc,
+            fields={
+                'Used CPU': {'per_time': 'Hourly', 'variable_rate': '3'},
+                'Used Disk I/O': {'per_time': 'Hourly', 'variable_rate': '2'},
+                'Used Memory': {'per_time': 'Hourly', 'variable_rate': '2'}
+            }
+        )
+        storage = appliance.collections.storage_rates.create(
+            description=desc,
+            fields={
+                'Used Disk Storage': {'per_time': 'Hourly', 'variable_rate': '3'}
+            }
+        )
+    except Exception as ex:
+        pytest.fail(
+            'Exception while creating compute/storage rates for chargeback report tests. {}'
+            .format(ex)
+        )
+
+    yield desc
+
+    for entity in [compute, storage]:
```

Comment:
**Question** Why not do the same thing in the above `new_compute_rate` fixture?

Reply:
can, was making change in these lines from first review.

Reply:
Actually, not sure what you mean here. This is the `new_compute_rate` fixture.

Reply:
discussed offline, talking about another test module. used same pattern there too.